### PR TITLE
feat(bundles): Add bundle endpoints

### PR DESCRIPTION
**Pull request by BinamB**
Jira Ticket: [PXP-5249](https://ctds-planx.atlassian.net/browse/PXP-5249)

### New Features
- Added Bundle endpoints.
- Added new table to store bundles.
- `\index\` endpoint resolves bundle ids.
- Added `form` parameter top `\index\` (without guid) to list either bundles, indexd records or both.
- DRS endpoint supports bundles.
- DRS endpoint supports `expand` parameter. 
- Added tests for bundles.
- Updated tests for DRS and Indexd

### CODE REVIEW

File: tests/test_setup.py [hunk 1]
```diff
@@ -35,6 +35,15 @@
         (1, "url", "VARCHAR", 1, None, 1 if OLD_SQLITE else 2),
     ],
     "index_schema_version": [(0, "version", "INTEGER", 1, None, 1)],
+    "drs_bundle_record": [
```
[the remark this answers is not in the document]
Comment (jiaqi216) [reply tests/test_setup.py:38]: alias etc - I think they were listed in my slides

File: indexd/drs/blueprint.py [hunk 2]
```diff
@@ -16,15 +16,54 @@ def get_drs_object(object_id):
...
     Returns a specific DRSobject with object_id
     """
+    expand = True if flask.request.args.get("expand") == "true" else False
+
     ret = blueprint.index_driver.get(object_id)
 
-    return flask.jsonify(indexd_to_drs(ret)), 200
+    data = (
+        indexd_to_drs(ret, expand=False, list_drs=False)
+        if not flask.request.args.get("expand")
```
Comment (jiaqi216) [inline indexd/drs/blueprint.py:25]: I'm a bit confused on this logic - why is there a different function call if the `expand` param is not available?

File: indexd/index/blueprint.py [hunk 3]
```diff
@@ -523,6 +549,123 @@ def version():
...
+
+    name = flask.request.json.get("name")
+    bundles = flask.request.json.get("bundles")
+    bundle_id = flask.request.json.get("bundle_id")
+    size = flask.request.json.get("size") if flask.request.json.get("size") else 0
+
+    if len(bundles) == 0:
+        raise UserError("Bundle data required.")
+
+    if len(bundles) != len(set(bundles)):
```
[the remark this answers is not in the document]
Comment (BinamB) [reply indexd/index/blueprint.py:658]: Did you mean, why we wouldn't support a bundle without any contents?

[the remark this answers is not in the document]
Comment (jiaqi216) [reply indexd/index/blueprint.py:658]: yeah we shouldn't support a bundle without contents

File: indexd/index/blueprint.py [hunk 4]
```diff
@@ -570,6 +597,143 @@ def version():
...
+    )
+
+    if len(bundles) == 0:
+        raise UserError("Bundle data required.")
+
+    if len(bundles) != len(set(bundles)):
+        raise UserError("Duplicate GUID in bundles.")
+
+    if bundle_id in bundles:
+        raise UserError("Bundle referes to itself.")
```
Comment (jiaqi216) [inline indexd/index/blueprint.py:662]: nit: refers

File: openapis/swagger.yaml [hunk 5]
```diff
@@ -137,6 +139,12 @@ paths:
       description: ''
       operationId: listEntries
       parameters:
+        - name: form
+          in: query
+          description: specify wheather you want to list bundle, objects or both.
```
Comment (jiaqi216) [inline openapis/swagger.yaml:144]: nit: whether 
Also what's the default? Is it just all?

File: openapis/swagger.yaml [hunk 6]
```diff
@@ -1114,7 +1134,117 @@ paths:
...
+      summary: Create a bundle.
+      description: ''
+      consumes: 
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - in: body
+          name: body
+          description: Body takes in GUID, bundle_name, checksum, list of bundles to bundle, and size. Only bundls bundles is required, rest will be created by indexd if not provided.
```
Comment (jiaqi216) [inline openapis/swagger.yaml:1150]: > Only bundls bundles
not sure what this means

File: openapis/swagger.yaml (hunk 5 above)
Comment (BinamB) [reply openapis/swagger.yaml:144]: it defaults to object for the `index` endpoint

**Pull request merged**